Added os_*_manage modules

retr0h opened the pull request:
These modules are used by the operator of OpenStack when deploying a cloud.  We would like to incorporate these into core, rather than fragmenting the community with various repos performing similar functionality.

Moving functionality from openstack-ansible/openstack-ansible-modules into core.

retr0h committed:
Added heat_manage module

This module is used by the operator when deploying the openstack heat service.
At the moment it only supports the `db_sync` option, but we can further build
on this as necessary.

Moving functionality from openstack-ansible/openstack-ansible-modules into core
to prevent fragmentation.

emonty commented:
@retr0h could you rename this prefixing with os_ to match the agreed naming scheme.

Also - I know this is a direct port in from os-ansible-modules (thanks for starting on that!) Do you envision this module growing the need to talk to the heat API for admin functions? Or is this purely about tasks needed to be performed on the host like db-sync?

retr0h commented:
@emonty prefix the filename to `os_` or the module's usage?

I would think this module will replicate the functionality of `heat-manage`.

retr0h committed:
Added the remaining *-manage modules

Pulled in the remaining *-manage modules from
openstack-ansible/openstack-ansible-modules.

retr0h committed:
Prefixed with os_ per @emonty's feedback

abadger commented:
@retr0h Hey, looks like @emonty had some questions about this set of modules.  Did the conversation go to a mailing list and then get resolved?

omgjlk commented:
These all need a license statement in the files.

omgjlk commented:
Instead of importing and running subprocess in each module, they should take advantage of the `run_command` function in the basic module_utils code within Ansible. https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1462

omgjlk commented:
When bringing these in, the docs should be updated to get rid of the action: syntax, and instead just use:

```
- name: name of task
  module_name:
    args_go_here: value
    other_args: other_value
```

gregdek commented:
@retr0h Please make the suggested revisions suggested by @j2sol. When you're done, please add a comment with the text "ready_for_review" and we will continue with the review process.  Thanks!
